test/python/test_substrait.py

```diff
@@ -15,3 +16,215 @@ def test_roundtrip_substrait(require):
 
     pd.testing.assert_series_equal(query_result.df()["i"], expected)
 
+
+def test_select_with_values(require):
+    connection = require('substrait')
+    query_result = execute_via_substrait(connection, "SELECT * FROM (VALUES (0),(1),(2),(3),(4),(5),(6),(7),(8),(9))")
+    expected = pd.Series(range(10), name="col0", dtype="int32")
+    pd.testing.assert_series_equal(query_result.df()["col0"], expected)
+
+
+def test_ctas_with_values(require):
+    connection = require('substrait')
+    query = "CREATE TABLE t1 AS SELECT * FROM (VALUES ('john', 25), ('jane', 21)) AS t(name, age)"
+    execute_via_substrait(connection, query)
+    expected = pd.Series(["john", "jane"], name="name", dtype="object")
+
+    query_result = execute_via_substrait(connection, "SELECT * FROM t1")
+    pd.testing.assert_series_equal(query_result.df()["name"], expected)
+
+
+def test_ctas_with_select_columns(require):
+    connection = require('substrait')
+    create_employee_table(connection)
+    _ = execute_via_substrait(connection, """CREATE TABLE employee_salaries AS 
+        SELECT name, salary FROM employees""")
+
+    expected = pd.DataFrame({"name": ["John Doe", "Jane Smith", "Alice Johnson", "Bob Brown", "Charlie Black"],
+                             "salary": [120000.0, 80000, 50000, 95000, 60000]})
```

Comment:
imo, since salary is decimal (10,2), we should mention all expected salary values as scale 2 i.e. 
[120000.00, 80000.00, 50000.00, 95000.00, 60000.00]
or may be add one of the salary as non zero scale value to make it more clear